affinity/cpp-23/d1795r1.md

```diff
@@ -124,11 +132,123 @@ As a result of the above this paper may also:
 * Propose a lifetime model for execution agents.
 * Propose some additions to the C\+\+ machine model to facilitate describing these additional properties.
 
-# Suggested straw polls
+# 6. Proposal
+
+## Header `<system>` synopsis
+
+```cpp
+namespace std {
+namespace experimental {
+
+/* system_topology */
+
+class system_topology {
+
+  system_topology() = delete;
+
+};
+
+/* system_resource */
+
+class system_resource {
+
+  system_resource() = delete;
+
+};
+
+/* traverse_topology */
+
+template <class T>
+ranges::view<system_resource> traverse_topology(const system_topology &, const T &) noexcept;
```

Comment:
See my previous comments.  It might be good to lay out the copy vs. view approaches as alternatives for voting.

Reply:
We could discuss it, but given that you can trivially collect the view into a vector if you want to keep a copy, is there a strong argument to be made for defaulting to copy?

Reply:
@trws The issue with views is their lifetime.  If we let the system topology change, then what happens to the view?  Do we make the implementation store a snapshot of the "last accessed topology"?

Reply:
As part of the system_topology object? I'd say yes.